src/nvim/api/vim.c

```diff
@@ -1679,21 +1465,6 @@ void nvim_del_keymap(uint64_t channel_id, String mode, String lhs, Error *err)
   nvim_buf_del_keymap(channel_id, -1, mode, lhs, err);
 }
 
-/// Gets a map of global (non-buffer-local) Ex commands.
-///
-/// Currently only |user-commands| are supported, not builtin Ex commands.
-///
-/// @param  opts  Optional parameters. Currently only supports
-///               {"builtin":false}
-/// @param[out]  err   Error details, if any.
-///
-/// @returns Map of maps describing commands.
-Dictionary nvim_get_commands(Dict(get_commands) *opts, Error *err)
```

Comment:
if we really don't like this in vim.c, how about renaming vimscript.c to editor.c and put most of the Ex/func stuff there?

Though I would still lean towards keeping it in vim.c. Basically vim.c could be "vim things" and most other modules are nvim things like lua, rpc, exmarks, etc.

Reply:
I think then "vimscript" would become too big. I'd regard ex/user commands as a separate feature as in (1) when the user does :%sub/foo/bar or (2) UserCommand where this command is defined in lua, then vimscript/eval features are not really involved. Though I suppose vimscript.c could be renamed eval.c, and also include nvim_x_set_var as these also use eval code

Reply:
>  Basically vim.c could be "vim things" and most other modules are nvim things like lua, rpc, exmarks, etc.

Given deep involvement of lua into pretty much all "vim things", such as commands mappings etc etc, I don't think this makes sense. The api in its entirety is owned by us. "vim.c for vim things" sounds like yet another misc1.c to me.

Reply:
Too small/granular is usually worse than too big, though. Can we wait to see if it actually becomes a problem?

Reply:
I don't think anything is becoming too granular either. Each new module in this PR is pretty big. It would be "wait and see" on that part as well.